### PR TITLE
Optional filename tag for retrieveData

### PR DESCRIPTION
**Pull request by pfuehrlich-pik**
I applied some auto-formatting, so if actual changes are hard to see only look at commits after the initial formatting.
fullXYZ functions may now `return(list(tag = "mytag"))`, which will result in a filename like rev0_h12_test_mytag.tgz (with mytag as suffix)

### CODE REVIEW

**Comment by k4rst3ns** [comment]
I am unsure, is there still a hash in the tgz-file name summarizing the settings of the full*-function call?
so that something like rev0_h12_test_mytag_<hash_including_all_parameters_of_full*_call>.tgz
I would be happy to get an info about what the name includes at the moment and what the changes introduces at the example of the fullCELLULARMAGPIE from mrmagpie.

**Comment by pfuehrlich-pik** [comment]
Yes the argument hash is still there. It is omitted if no arguments are passed to the fullXYZ method, that's why it is missing in my example. The file name is the same as before, here is the code where it is assembled:
```
collectionname <- paste0(
    "rev", rev, dev, "_", regionscode, "_", argsHash, tolower(model),
    ifelse(getConfig("debug") == TRUE, "_debug", "")
  )
```
With this PR a tag is appended to the end of the filename (but before .tgz) if the fullXYZ method ends with `return(list(tag="mytag"))`. Here's the corresponding code:
```
    collectionname <- paste0(collectionname,
                             ifelse(is.list(returnedValue) && exists("tag", returnedValue),
                                    paste0("_", returnedValue$tag),
                                    ""))
```
So we have a filename like rev0_h12_test_1234567_mytag.tgz, where 1234567 is the argsHash.
For fullCELLULARMAGPIE nothing will change, unless it returns list(tag = "anything").

**Comment by pfuehrlich-pik** [comment]
The tag is not used in the cachefolder name. However, caching is not applied to fullXYZ functions, but rather to the calc functions called from fullXYZ. These are cached as before, the tag does not show up here, but the cache files are still valid when running the fullXYZ again with a different tag.

**Review by k4rst3ns — APPROVED** [review]
From user side no issue just praise for these changes.

File: R/retrieveData.R [hunk 1]
```diff
@@ -1,132 +1,175 @@
...
+  collectionname <- paste0(
+    "rev", rev, dev, "_", regionscode, "_", argsHash, tolower(model),
+    ifelse(getConfig("debug") == TRUE, "_debug", "")
+  )
+
+  .characterInCollectionname <- function(regexCharacter) {
+    return(grepl(regexCharacter, collectionname, fixed = TRUE))
+  }
+  if (any(unlist(lapply(strsplit("[](){}*+?|^$.\\", "")[[1]], .characterInCollectionname)))) {
+    warning(paste0("At least one of the regex characters [](){}*+?|^$.\\ appeared in collectionname",
```
Comment (tscheypidi) [inline R/retrieveData.R:105]: can you add a test which triggers this new warning?

**Review by tscheypidi — CHANGES_REQUESTED** [review]

**Review by tscheypidi — APPROVED** [review]

**Pull request merged**